### ISSUE TITLE
DCD or other tools started should respect spaces in location paths

### ISSUE THREAD

**Issue by webloft** [intellij-dlanguage/intellij-dlanguage#590]
Tested on windows:

DCD cannot find import locations because some whitespace in location paths are not properly escaped.

**Comment by webloft** [comment]
This also affects any other escape characters - eg. "\\tmp" passed as "\mp" to debugger

**Comment by SingingBush** [comment]
The GDB code is fairly different to the other tools and will probably be fixed by overhauling it completely for #598. For this ticket it's best to focus on DCD. @webloft please could you test this with code in the develop branch and if it's not working provide more details so that it can be reproduced. Please include text that was used when the plugin called dcd and what you expected it to be.

**Comment by webloft** [comment]
I'm afraid I'm not familiar with Java enough to build it on my own.

I think the problem was with -I argument of dcd server. If my working directory cointains a space it was passed as two arguments, resulting in an invalid directory and one unknown argument error. 

However I haven't use it in a while and I'm not sure. I assume that the working directory were passed automatically, now the server starts without any argument(?) Need to pass the -I argument via flags option. But this works and issue seems to be fixed in the 1.26 release.

**Comment by SingingBush** [comment]
ok, there are some changes in the latest 1.26 release that should help with some areas such as the gdb args. Probably further changes needed as well though. Could you try with the latest version and give examples of any areas that are broken in this regard as the various tools have code that is specific to each so it'll need treating as independent pieces of work. Getting an example of the output you are seeing in the console compared to what it should be would be very helpful so that I can track down the specific classes that cause problems.

**Comment by SingingBush** [comment]
this may be solved in the constructor for _DCDCompletionServer_ by swapping the line:

```
this.workingDirectory = lookupWorkingDirectory();
```

to instead be:

```
this.workingDirectory = ModuleUtil.getModuleDirPath(module);
```

will need some testing though.

**Comment by SingingBush** [comment]
@webloft I'd like to get this out with the next release. Would you be happy to test the latest change on the develop branch. Just check out the develop branch and run `./gradlew runIde` from the project root. It'll spin up a sandboxed Intellij IDEA and then you just need to setup dmd and dcd as usual.

**Comment by SingingBush** [comment]
paths on Windows for dcd have been fixed as part of #633, see https://github.com/intellij-dlanguage/intellij-dlanguage/commit/19ba4ee7b789abd1c58f293ed8ad3c9e49a1ca03 which makes use of `PathUtil::toSystemDependentName`

I'll take a look at how GDB is run and see about putting in a similar fix. I could really do with help testing so checking out the develop branch and running `./gradlew runIde` and trying it out would be appreciated
